pyerrors/input/openQCD.py

```diff
@@ -343,3 +345,209 @@ def _read_array_openQCD2(fp):
+                    t = fp.read(8 * tmax * (nn + 1))#wegwerfen, Ysl
+                    t = fp.read(8 * tmax * (nn + 1))#NICHT wegwerfen, das ist Qsl, das, was wirr wollen
+                    #unpack the array of Qtops, on each timeslice t=0,...,tmax-1 and the 
+                    #measurement number in = 0...nn (see README.qcd1)
+                    tmpd = struct.unpack('d' * tmax * (nn + 1), t)
+                    Q.append(tmpd)
+                    i += 1*steps
+                #print(tmp)
+                #set step by reading all entries, then set stepsize, then check if everything is there
+                #make a dtr_config param, which is checked against difference...
+                #difference != step
+                #len(set(difference)) == 1
+                #!!!also implement the idl stuff for everything...
+        print(len(Q))
+        print('max_t:', dn * (nn) * eps)
+        
+        t_aim = (c * L) ** 2 / 8
+
+        print('t_aim:', t_aim)
+        index_aim = round(t_aim / eps / dn)
+        print('index_aim:', index_aim)
+
+
+        Q_sum = []
+        for i, item in enumerate(Q):
+            Q_sum.append([sum(item[current:current + tmax]) for current in range(0, len(item), tmax)])
+        print(len(Q_sum))
+        print(len(Q_sum[0]))
+        Q_round = []
+        for i in range(len(Q) // dtr_cnfg):
+            Q_round.append(round(Q_sum[dtr_cnfg * i][index_aim]))
+
+        replica = len(files)
+
+        tmp = []
+        for q in Q_round:
+            if full:
+                tmp.append(q)
+            else:
+                if int(q) == target:
+                    tmp.append(1.0)
+                else:
+                    tmp.append(0.0)
+        
+        truncated_file = file[:-7] #as seen in previous examples, this could lead to some weird behaviour... maybe -7 fixes this.
+        print(truncated_file)
+        try:
+            idx = truncated_file.index('r')
+        except:
+            if not "names" in kwargs:
+                raise Exception("Automatic recognition of replicum failed, please enter the key word 'names'.")
+        
+        #print(truncated_file[idx:])
+        # this might be a quite fishy way to find out which replicum we are actually talking about...
+        if "r_start" in kwargs:
+            tmp = tmp[r_start[int(truncated_file[idx+1:])-1]:]
+        if "r_stop" in kwargs:
+            tmp = tmp[:r_stop[int(truncated_file[idx+1:])-1]]
+        if "ens_name" in kwargs:
+            ens_name = kwargs.get("ens_name")
+        else:
+            ens_name = truncated_file[:idx]
+        #keyword "names" prevails over "ens_name"
+        if not "names" in kwargs:
+            rep_names.append(truncated_file[:idx] + '|' + truncated_file[idx:])
+        else:
+            names = kwargs.get("names")
+            rep_names = names
+        deltas.append(np.array(tmp))
+
+        
+    result = Obs(deltas, rep_names)
```

Comment:
Ideally, the `Obs` object would also obtain the `idl` of the data.

Reply:
Done, hopefully in the right way.